compiler/rustc_errors/src/lib.rs

```diff
@@ -1224,9 +1224,8 @@ impl DiagCtxtInner {
     /// Emit all stashed diagnostics.
     fn emit_stashed_diagnostics(&mut self) -> Option<ErrorGuaranteed> {
         let has_errors = self.has_errors();
-        let diags = self.stashed_diagnostics.drain(..).map(|x| x.1).collect::<Vec<_>>();
         let mut reported = None;
-        for diag in diags {
+        for (_, diag) in std::mem::take(&mut self.stashed_diagnostics).into_iter() {
```

Comment:
The `drain` kept the allocation around instead of freeing it, but this is not hot

Reply:
I could do `for (_, diag) in self.stashed_diagnostics.drain(..)`. Like you say, perf is irrelevant, but that would be more concise.

Reply:
That doesn't work.
```
error[E0499]: cannot borrow `*self` as mutable more than once at a time
    --> compiler/rustc_errors/src/lib.rs:1244:33
     |
1228 |         for (_, diag) in self.stashed_diagnostics.drain(..) {
     |                          ----------------------------------
     |                          |
     |                          first mutable borrow occurs here
     |                          first borrow later used here
...
1244 |             let reported_this = self.emit_diagnostic(diag);
     |                                 ^^^^ second mutable borrow occurs here
```
I can't think of anything better, I'll leave it using `std::mem::take`.

Reply:
oh yea, that's fine